### PR TITLE
Add non-functional requirements to Developer Guide

### PR DESCRIPTION
**Pull request by rayyan35p**

### CODE REVIEW

**Review by aslam341 — COMMENTED** [review]
One of the non-functional requirement is unclear to me, the rest seems good.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -315,6 +315,9 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4.  The system should respond within two seconds.
```
[the remark this answers is not in the document]
Comment (rayyan35p) [reply docs/DeveloperGuide.md:318]: I personally thought 2 seconds was reasonable. Perhaps I should specify the timing which we define as sluggishness (More than 2 seconds)?

[the remark this answers is not in the document]
Comment (antonTan96) [reply docs/DeveloperGuide.md:318]: I think we can merge 4 into 2:
`... without a noticeable sluggishness (at most 2 seconds of wait time per command) ...`

[the remark this answers is not in the document]
Comment (rayyan35p) [reply docs/DeveloperGuide.md:318]: Noted with thanks! Will update for better clarity.

[the remark this answers is not in the document]
Comment (timleow) [reply docs/DeveloperGuide.md:318]: What if we just leave the duration out for now? We haven't tested our system to the point where it gets sluggish yet.

**Review by antonTan96 — APPROVED** [review]
LGTM!

**Pull request merged**